### PR TITLE
[ASM] - EXPANDER 4377 - ServiceNow ITSM Enrichment

### PR DESCRIPTION
**Pull request by BigEasyJ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
[EXPANDR-4377](https://jira-dc.paloaltonetworks.com/browse/EXPANDR-4377)

## Description
Create a new playbook to query ServiceNow ITSM for service owners.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  '27':
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:356]: @BigEasyJ , in this task (or a separate task) we need to make sure SNOW-ITSM integration is enabled or there is no reason to continue.

**Review by johnnywilkes — CHANGES_REQUESTED** [review]

**Review by johnnywilkes — APPROVED** [review]
@BigEasyJ , thanks for the change.

**Comment by capanw** [comment]
LGTM

**Comment by sapirshuker** [comment]
@ShirleyDenkberg , Can you please review?

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    isautoswitchedtoquietmode: false
+  "25":
+    id: "25"
+    taskid: d3007e79-c33c-4f6f-83f3-e7ee98dfb3b2
+    type: regular
+    task:
+      id: d3007e79-c33c-4f6f-83f3-e7ee98dfb3b2
+      version: -1
+      name: Build ServiceNow query - description
+      description: Set the beginning of a key used for querying ServiceNow using SN descriptions and a timestamp.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:241]:
```suggestion
      description: Set the beginning of a key used for querying ServiceNow using ServiceNow descriptions and a timestamp.
```

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+    isautoswitchedtoquietmode: false
+  "26":
+    id: "26"
+    taskid: 962733bf-859d-4823-8d39-478d4fb092e8
+    type: regular
+    task:
+      id: 962733bf-859d-4823-8d39-478d4fb092e8
+      version: -1
+      name: Build ServiceNow query - short description
+      description: Set the remaining query value with SN short descriptions for querying ServiceNow.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:302]:
```suggestion
      description: Set the remaining query value with ServiceNow short descriptions for querying ServiceNow.
```

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+    isautoswitchedtoquietmode: false
+  "27":
+    id: "27"
+    taskid: 449135ee-dbef-48c5-8e00-597ce6f55cf0
+    type: condition
+    task:
+      id: 449135ee-dbef-48c5-8e00-597ce6f55cf0
+      version: -1
+      name: Were search terms provided?
+      description: Conditional to check if values were provided to the search terms input or not.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:366]:
```suggestion
      description: Check if values were provided to the search terms input or not.
```

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+    isautoswitchedtoquietmode: false
+  "32":
+    id: "32"
+    taskid: c9c5b720-7cd3-406e-8a2e-3af77876dc8b
+    type: condition
+    task:
+      id: c9c5b720-7cd3-406e-8a2e-3af77876dc8b
+      version: -1
+      name: Were any assignees found?
+      description: Conditional to check if any assignees were found in the assigned_to property of retrieved ServiceNow tickets.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:410]:
```suggestion
      description: Check if any assignees were found in the assigned_to property of retrieved ServiceNow tickets.
```

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+        "x": 120,
+        "y": -1140
+      }
+    }
+  }
+inputs:
+- key: search_terms
+  value: {}
+  required: true
+  description: Search terms to be used in the ServiceNow ITSM query search
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment.yml:529]:
```suggestion
  description: Search terms to be used in the ServiceNow ITSM query search.
```

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment_README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+This playbook uses the following sub-playbooks, integrations, and scripts.
+
+### Sub-playbooks
+
+This playbook does not use any sub-playbooks.
+
+### Integrations
+
+* ServiceNow v2
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_ServiceNow_ITSM_Enrichment_README.md:13]:
```suggestion
ServiceNow v2
```

**Comment by ShirleyDenkberg** [comment]
@sapirshuker When there is only one item in a list, no hyphen is needed.

@johnnywilkes @capanw @melamedbn @sapirshuker Doc review completed.

**Comment by sapirshuker** [comment]
Hey, @BigEasyJ
Please fix the doc review comments and let me know when the PR is ready for review.

**Comment by BigEasyJ** [comment]
@sapirshuker updated and ready

**Comment by sapirshuker** [comment]
Build process failed due an known issue: [CIAC-6419](https://jira-hq.paloaltonetworks.local/browse/CIAC-6419)

**Review by sapirshuker — APPROVED** [review]

**Comment by sapirshuker** [comment]
Hey, @BigEasyJ 
please resolve the conflicts.

**Comment by sapirshuker** [comment]
Build process failed due an known issue: [CIAC-6419](https://jira-hq.paloaltonetworks.local/browse/CIAC-6419)

**Review by sapirshuker — APPROVED** [review]

**Pull request merged**